### PR TITLE
Dfp

### PR DESCRIPTION
**Pull request by dwick**
:eyeglasses: @ajacksified @curioussavage 

Might need to do some more with tagging, but this should work for the mean time.

### CODE REVIEW

File: src/lib/frames.es6.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    customEvent.source = e.source;
+
+    global.dispatchEvent(customEvent);
+  } catch (x) {}
+}
+
+
+
+function compileOriginRegExp(origins) {
+  return new RegExp('^http(s)?:\\/\\/' + origins.join('|'), 'i');
```
Comment (JordanMilne) [inline src/lib/frames.es6.js]: Should this have a `$` anchor at the end of the regex as well?

Comment (dwick) [inline src/lib/frames.es6.js]: good catch

File: src/config.es6.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ var config = {
 
   loginPath: process.env.LOGIN_PATH || '/oauth2/login',
 
-  adsPath: process.env.ADS_PATH ||  'https://www.reddit.com/api/request_promo.json',
+  adsSlot: process.env.ADS_PATH || '/32173961/mw_card_test_1',
```
Comment (ajacksified) [inline src/config.es6.js]: intentional?

File: src/lib/frames.es6.js [hunk 3]
```diff
@@ -0,0 +1,210 @@
+'use strict';
+
+var WILDCARD = '*';
+var ALLOW_WILDCARD = '.*';
+var RE_WILDCARD = /\*/;
+var DEFAULT_MESSAGE_NAMESPACE = '.postMessage';
+var DEFAULT_POSTMESSAGE_OPTIONS = {
```
Comment (ajacksified) [inline src/lib/frames.es6.js]: might as well make all these `const` instead of `var`

File: src/lib/frames.es6.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+   */
+  receiveMessage: function (source, type, callback, context) {
+    if (typeof source === 'string') {
+      context = callback;
+      callback = type;
+      type = source;
+      source = null;
+    }
+
+    if (!/\..+$/.test(type)) {
```
Comment (ajacksified) [inline src/lib/frames.es6.js]: all of the other regexes are variables; put this one in a const too?

File: src/views/components/Ad.jsx [hunk 5]
```diff
@@ -4,20 +4,92 @@ import superagent from 'superagent';
...
+        googletag.pubads().enableSingleRequest();
+        googletag.enableServices();
+      });
+
+      googletag.cmd.push(function() {
+        googletag.display(adId);
+      });
+
+      setTimeout(() => {
+        reject('timeout');
```
Comment (ajacksified) [inline src/views/components/Ad.jsx]: can you be more specific? `'ad creative load timeout'` or something.

**Comment by ajacksified** [comment]
A few nits / questions, but otherwise good to go. :nail_care:

File: src/views/components/Ad.jsx [hunk 6]
```diff
@@ -4,20 +4,92 @@ import superagent from 'superagent';
...
+        '/api/dfp/link.json';
+    var headers = {};
+
+    if (props.token) {
+      headers.authorization = 'bearer ' + props.token;
+    };
+
+    return new Promise((resolve, reject) => {
+      superagent
+        .post(url)
```
Comment (JordanMilne) [inline src/views/components/Ad.jsx]: If the `websafe_json` encoding will mess things up here you should add a `raw_json=1` parameter to the URL or POSTDATA.

Comment (dwick) [inline src/views/components/Ad.jsx]: :+1:

File: src/lib/frames.es6.js (hunk 4 above)
Comment (dwick) [inline src/lib/frames.es6.js]: i removed the regexes that weren't reused from the constants

**Comment by dwick** [comment]
:haircut:

**Comment by ajacksified** [comment]
Looks good; :fish: pending testing on staging and DFP-readiness on the reddit.com side

**Comment by dwick** [comment]
not fantastic enough.

**Comment by ajacksified** [comment]
No, @dwick. The problem was that it was *too* fantastic.

**Pull request closed without merging**